nes-systests/systest/src/SystestResultCheck.cpp

```diff
@@ -269,19 +270,21 @@ struct QueryResult
     std::vector<std::string> result;
 };
 
-std::optional<QueryResult> loadQueryResult(const NES::Systest::SystestQuery& query)
+std::optional<QueryResult> loadQueryResult(const std::filesystem::path& resultFilePath)
 {
-    NES_DEBUG("Loading query result for query: {} from queryResultFile: {}", query.queryDefinition, query.resultFile());
-    std::ifstream resultFile(query.resultFile());
+    NES_DEBUG("Loading query result from: {}", resultFilePath);
+    std::ifstream resultFile(resultFilePath);
     if (!resultFile)
     {
-        throw NES::UnknownException("Failed to open result file: {}", query.resultFile());
+        NES_ERROR("Failed to open result file: {}", resultFilePath);
+        return std::nullopt;
     }
 
     QueryResult result;
     std::string firstLine;
     if (!std::getline(resultFile, firstLine))
     {
+        NES_WARNING("Result file is empty: {}", resultFilePath);
```

Comment:
Shouldn't that be an error? The FileSink should at least write the schema in its `start` method.